src/FSharp.Core/prim-types.fs

```diff
@@ -5206,6 +5217,29 @@ namespace Microsoft.FSharp.Core
                 if value = unsafeDefault<'T> then ""
                 else let x = (# "" value : IFormattable #) in defaultIfNull "" (x.ToString(null, CultureInfo.InvariantCulture))
 
+        [<CompiledName("ToString")>]
+        let inline string (value: 'T) = 
+             defaultString value
+
+             // Only compilers that understand `when 'T : SupportsWhenTEnum` will understand `when 'T : Enum`.
+             when 'T : SupportsWhenTEnum =
+                (
+                    let inline string (value : 'T) =
+                        defaultString value
+
+                        // Special handling for enums whose underlying type is a signed integral type.
+                        // The runtime value may be outside the defined members of the enum, and the negative sign may be overridden.
+                        when 'T : Enum  = let x = (# "" value : 'T #) in x.ToString() // Use 'T to constrain the call to the specific enum type.
```

Comment:
This clause is when working with generic `Enum` code - please add a comment linking that to the specific source code constructs this handles.

Reply:
Maybe add `WhenTyparIsEnum` in the comment - I believe that's what it is, right?

Reply:
d5dc4a5